### DISCUSSION TITLE
Does Tycho 3.0.1 supports JUnit 5.9 ?

### DISCUSSION THREAD

**Discussion by glhez** [eclipse-tycho/tycho#1924] [Q&A] [answered]
Hello,

I am trying to use JUnit 5.9.1 with Tycho 3.0.1, and I ran into this exception when running a test (which work in Eclipse):

>java.lang.AbstractMethodError: Receiver class org.junit.jupiter.engine.descriptor.TestTemplateExtensionContext does not define or inherit an implementation of the resolved method 'abstract org.junit.jupiter.api.extension.ExecutableInvoker getExecutableInvoker()' of interface org.junit.jupiter.api.extension.ExtensionContext.

The method `getExecutableInvoker()` was added in JUnit 5.9 which means that somehow, Tycho is building a classpath containing an older implementation of `TestTemplateExtensionContext` (< 5.9.0).

My META-INF was changed to reference the new OSGI bundles (I omitted some part):

- Before: `Require-Bundle: org.eclipse.jface.text,org.junit.jupiter.api,org.junit.jupiter.params,`
- After:  `Require-Bundle: org.eclipse.jface.text,junit-jupiter-api,junit-jupiter-params,`

In reading other issues mentioning JUnit 5.9, I have come to understand that JUnit 5.9 was more or less being supported here with Tycho 3.0.1 (https://github.com/eclipse-tycho/tycho/issues?q=is%3Aissue+junit+5.9).

But:

(1) I've made a quick scan of jar containing the `TestTemplateExtensionContext.class` based on the `osgi.bundles` property found in the `TEST-XXXX.xml` generated by surefire:

$M2_REPO/org/eclipse/tycho/org.eclipse.tycho.surefire.junit58withvintage/3.0.1/org.eclipse.tycho.surefire.junit58withvintage-3.0.1.jar
$M2_REPO/p2/osgi/bundle/junit-jupiter-engine/5.9.0/junit-jupiter-engine-5.9.0.jar

(2) I can see that in master branch there is a [junit59withvintage](https://github.com/eclipse-tycho/tycho/tree/master/tycho-surefire/org.eclipse.tycho.surefire.junit59withvintage) but it is targeted for a potential 4.0.0-SNAPSHOT. 

(3) If I stick to the old OSGI bundle name, I can see in the dependency collection (using mvn -X) that Tycho is selecting JUnit 5.8.2

Is there some additional task to perform to use JUnit 5.9.1 with Tycho 3.0.1 ?

**Comment by mickaelistria** [comment]
I don't think Tycho 3.0.1 entirely support JUnit 5.9; it may support some subset but there is dedicated runner configured for it.
Have you tried whether Tycho 4.0.0-SNAPSHOT works as expected with JUnit 5.9?

**Reply by glhez** [reply]
I just tried it, and yes it seems to work now.
I'll wait for it to be released then.

**Comment by Bananeweizen** [comment, marked as answer]
You probably experienced this bug https://github.com/eclipse-tycho/tycho/issues/1943. Upgrading to Tycho 3.0.3 fixes it.
